devmand/gateway/src/devmand/channels/cli/QueuedCli.h

```diff
@@ -36,27 +41,37 @@ class QueuedCli : public Cli {
    * Unbounded multi producer single consumer queue where consumer is not
    * blocked on dequeue.
    */
-  UnboundedQueue<QueueEntry, false, true, false> queue;
-  bool isProcessing = false; // only accessed from consumer thread
+  UnboundedQueue<QueueEntry, false, true, false>
+      queue; // TODO: investigate priority queue for keepalive commands
 
- public:
-  QueuedCli(
-      string _id,
-      shared_ptr<Cli> _cli,
-      const shared_ptr<Executor>& _parentExecutor);
-  QueuedCli() = delete;
-  ~QueuedCli() override;
-  QueuedCli(const QueuedCli&) = delete;
+  atomic<bool> isProcessing;
 
-  Future<string> executeAndRead(const Command& cmd) override;
-  Future<string> execute(const Command& cmd) override;
+  atomic<bool> shutdown;
+
+  QueuedCli(
+      string id,
+      shared_ptr<Cli> cli,
+      shared_ptr<Executor> parentExecutor);
 
- private:
   Future<string> executeSomething(
       const Command& cmd,
       const string& prefix,
       function<Future<string>()> innerFunc);
 
   void triggerDequeue();
+
+ public:
+  static std::shared_ptr<QueuedCli>
+  make(string id, shared_ptr<Cli> cli, shared_ptr<Executor> parentExecutor);
+
+  QueuedCli() = delete;
+
+  QueuedCli(const QueuedCli&) = delete;
```

Comment:
Is it necessary to delete this copy constructor if parent class Cli already deletes it ?
Also is it necessary to delete default (no arg) constructor if we have a custom one ? Just checking ... because in KeepaliveCli we don't delete constructors

Reply:
You're right, deleting those